### PR TITLE
Klingon and acclaim

### PR DESCRIPTION
**Pull request by tokeidlom**
Adding Klingon character sheet and acclaim support to 1e character sheet

### CODE REVIEW

File: src/lang/pt.json
Comment (tokeidlom) [inline src/lang/pt.json:1]: Would you like to review this before it gets merged @adambravo79 ?

Comment (adambravo79) [reply src/lang/pt.json:1]: Will do

File: src/lang/pt.json
Comment (adambravo79) [inline src/lang/pt.json:123]: might is better translated as: "Poder"

**Review by adambravo79 — CHANGES_REQUESTED** [review]
It is okay, just recommend to change the 'might' string to "Poder".

File: src/lang/pt.json
Comment (tokeidlom) [reply src/lang/pt.json]: Done, thank you!

File: src/module/actors/sheets/character-sheet.js
Comment (WesWedding) [inline src/module/actors/sheets/character-sheet.js:483]: Maybe this should live in a Handlebars file instead of being hardcoded here.

**Review by WesWedding — COMMENTED** [review]
There's a lot to review here!

----
Making sure that all fields are included from the printed sheet is an admirable goal but I feel that the Klingon variant is quite overwhelmingly detailed at this point.  To the point of making it actually hard to use, even.  Many of the fields aren't Klingon-specific, either, so hiding them solely on the Klingon setting seems wrong to me.

Perhaps a new section of inputs needs to be introduced at the very bottom of the sheet to where fields can live, so we're not pushing the Attribs, Disciplines, Belongings, etc lower and lower.

----
Acclaim/Reprimand and Reputation should probably be closer to the reputation bar, since they're so closely linked conceptually.  Making them on opposite sides of the fields introduces a spatial and conceptual separation.  

----
Putting all of those extra fields for Klingon stuff behind a Setting seems counterintuitive to me.

If I were a user and wanted to have a Klingon sheet I think I would expect to be able to just switch to a Klingon sheet using the normal Foundry workflow.
image

----
From a developer perspective, when a "if" block gets especially massive, that tends to be a code smell to me;  I start to look for a way to move all of that nested logic into a new function.

I feel like this extends to the HBS file, too.  Seems like a good case for using Handlebars partials when an `{{if}}` is wrapped around simply massive chunks of HTML.

----
Lots of hardcoded `width=X%` which, unfortunately, was done earlier in System development and we're sort of just continuing to expand as new fields come in.  Generally if there's a style that needs to be applied it should be done in CSS, so it's easier to find.  Something that probably deserves a cleanup later, I guess!

**Comment by tokeidlom** [comment]
I end up with a sheet that looks something like this:
image

**Comment by WesWedding** [comment]
Another brainstorm idea for managing space is that comes to mind is [collapsible ](https://www.w3schools.com/howto/howto_js_collapsible.asp )sections.   It might require some reworking, but you could have some collections of fields come pre-collapsed, but maybe that would be a win.  A downside of collapsible sections is that usually mean that the uncollapsed state is even taller than what it was before the fields got wrapped, though.

**Comment by WesWedding** [comment]
I like that grouping of rep at the bottom, looks slick.

**Comment by tokeidlom** [comment]
Alternatively there is another path we could take (these are a basic mock up as a proof on concept, not a finished product), I would need to know if this is a route we wanted to take before sinking more time into this as it'll be a massive job!
image
image
image
image

**Comment by WesWedding** [comment]
I wish there was a way to solicit user feedback on this!  Organizing things seems like a good spot for at least collecting feedback.

There's always tradeoffs to approaches and even though I favor Tabs over other approaches, we might as well acknowledge one downside that has come in in convo about tabs before which is: it can get a little disorienting having to switch tabs.  Especially in STA.

Unlike other some RPG systems, almost every Task including combat includes a broad spectrum of mechanics to consider.  And this system doesn't automate much, so the player has to keep track of all these mechanics every roll (and, in my GM experience, I have to guide them through it every time).

The list is long, especially in combat.  You have to review Attributes, Disciplines, Traits, Values, Focuses, Weapons, Armor (maybe more?).   So splitting Items from Values, Focuses, etc is actually a little bit annoying or even disorienting.  Or at least, I have occasionally found it disorienting when I'm a GM bouncing around multiple characters that might all be on different tabs.  Even a non-combat Task could require switching tabs if it involves inventory.

However, the upsides of Tabs are incredibly obvious.  You get a more compact sheet, you get to be more free with new fields, it means there is room to actually add some missing stuff like Milestones in the future, and so on.

**Comment by WesWedding** [comment]
I am personally in favor of the overall layout you're suggesting, it's extremely similar to something I wanted to do in STA Enhanced eventually but it was going to be a lot more work than just adding some tabs.  I think either DND5E or Pathfinder used a similar layout, where it has a sidebar with the primary stats.

I think maybe a "Community Feedback: Sheet Reorganizing" or maybe even something attention-getting like "Community Feedback: Character Sheet 2.0" issue in the Issue queue would be good idea, with mockups posted (without LCARS, too) just to see what people think before making a commitment to the work.

**Comment by tokeidlom** [comment]
I could move the species row into the bio tab which would mean all the items sheets could probably be put into one tab.

What's required for milestones? Would another text field similar to notes, but dedicated to milestones be adequate?

**Comment by WesWedding** [comment]
It depends entirely on how feature-rich the system goes, you know?  There is currently an unfinished Milestone item already in the system, so that could be built up a bit.  And then that leads to a conversation about what goes in the item...  A field for the value used?  A negative, positive, challenge indicator?  Track the reward used?  Flag it as a character arc?

Mission logs are also a key part of the Milestone system.  Should that be an Item? Where should we store those?

We shouldn't have that discussion in this thread, though.  This PR is about supporting Klingon and acclaim.  If the sheet redesign has to come first, then I feel like this PR should be put on hold and sheet redesign needs to go somewhere else.

**Pull request closed without merging**
